sql/core/src/test/scala/org/apache/spark/sql/sources/v2/DataSourceV2Suite.scala

```diff
@@ -392,6 +392,16 @@ class DataSourceV2Suite extends QueryTest with SharedSQLContext {
       }
     }
   }
+
+  test("SPARK-27411: DataSourceV2Strategy should not eliminate subquery") {
+    val t2 = spark.read.format(classOf[SimpleDataSourceV2].getName).load()
+    sql("create temporary view t1 (a int) using parquet")
+    val df = t2.where("i < (select max(a) from t1)")
+    val subqueries = df.queryExecution.executedPlan.collect {
+      case p => p.subqueries
+    }.flatten
+    assert(subqueries.length == 1)
```

Comment:
let's also check the result. I think this is a correctness bug?

Reply:
Sure, thanks.